onnxruntime/core/providers/cpu/tensor/pad.cc

```diff
@@ -247,6 +275,51 @@ Status PadCpuImpl<float>(OpKernelContext* ctx,
 
 template <>
 Status Pad<float>::Compute(OpKernelContext* ctx) const {
-  return PadCpuImpl<float>(ctx, pads_, slices_, mode_, value_);
+  // kOnnxDomain Pad opset >= 11 (Or) kMsDomain opset == 1
+  if (is_dynamic_) {
+    const Tensor& input_tensor = *ctx->Input<Tensor>(0);
+    size_t data_rank = input_tensor.Shape().NumDimensions();
+
+    const Tensor& pads_tensor = *ctx->Input<Tensor>(1);
+    const std::vector<int64_t>& pads_tensor_dims = pads_tensor.Shape().GetDims();
+    ORT_ENFORCE(pads_tensor.DataType() == DataTypeImpl::GetType<int64_t>(),
+                "Pads tensor should be an INT64 tensor");
+    ORT_ENFORCE(pads_tensor_dims.size() == 1 || (pads_tensor_dims.size() == 2 && pads_tensor_dims[0] == 1),
+                "Pads tensor should be a 1D tensor of shape [2 * input_rank] or a 2D tensor of shape [1, 2 * input_rank]");
+
+    const int64_t* pads_tensor_raw_data = pads_tensor.template Data<int64_t>();
+    size_t pads_size = static_cast<size_t>(pads_tensor.Shape().Size());
+    ORT_ENFORCE(pads_size == 2 * data_rank,
+                "Pads tensor size should be equal to twice the input dimension count ");
+
+    std::vector<int64_t> pads;
+    pads.reserve(2 * data_rank);
+    for (size_t i = 0; i < pads_size; ++i) {
+      pads.push_back(pads_tensor_raw_data[i]);
+    }
+
+    // Separate out any negative pads into the slices array
+    std::vector<int64_t> slices(pads.size(), 0);
```

Comment:
PR feedback addressed:

1) bought back `slices` to the original state
2) Removed the unnecessary line in `pad.h`